### PR TITLE
Benji/ios 489 start conversation on first unread

### PR DESCRIPTION
**Pull request by Benjamindodgson**
There are a few tasks in here I rolled up into this.

### CODE REVIEW

File: Shared/Managers/UserDefaultsManager.swift [hunk 1]
```diff
@@ -13,13 +13,18 @@ class UserDefaultsManager {
...
+        case shouldShowInvestUpsell
+        case shouldShowGroupsUpsell
     }
 
     static func update(key: Key, with value: Any) {
         UserDefaults.standard.setValue(value, forKey: key.rawValue)
     }
 
-    static func getBool(for key: Key) -> Bool {
+    static func getBool(for key: Key, defaultValue: Bool = true) -> Bool {
```
Comment (Benjamindodgson) [inline Shared/Managers/UserDefaultsManager.swift:24]: Added a close button to the cards that save a local bool

File: Shared/Models/MicroAnimation.swift [hunk 2]
```diff
@@ -9,8 +9,11 @@
 import Foundation
 
 enum MicroAnimation: String {
-    case checkbox = "checkbox"
     case error = "error"
     case faceScan = "face_scan"
     case loading = "loading"
+    
+    case alertCircle = "alertCircle"
```
Comment (Benjamindodgson) [inline Shared/Models/MicroAnimation.swift:16]: Updated to reflect the animations

File: Shared/Models/Parse Subclasses/Reservation.swift [hunk 3]
```diff
@@ -108,12 +108,12 @@ extension Reservation: UIActivityItemSource {
 
     var message: String? {
         //guard let link = self.link else { return nil }
-        return "RSVP code: \(String(optional: self.objectId))\nClaim your RSVP by tapping 👇\n\(Reservation.domainURL), and then entering the code."
+        return "Get the Jibber app so we can communicate with empathy.\nRSVP by tapping 👇\n\(Reservation.domainURL)\nand then enter this code:\n\(String(optional: self.objectId))"
```
Comment (Benjamindodgson) [inline Shared/Models/Parse Subclasses/Reservation.swift:111]: Better invite message

File: Shared/Notifications/UserNotificationManager.swift [hunk 4]
```diff
@@ -130,7 +130,7 @@ class UserNotificationManager: NSObject {
                 var identifiers: [String] = []
                 var badgeCount = self.application?.applicationIconBadgeNumber ?? 0
                 delivered.forEach { note in
-                    if note.request.identifier == message.id {
+                    if note.request.content.messageId == message.id {
```
Comment (Benjamindodgson) [inline Shared/Notifications/UserNotificationManager.swift:133]: Finding the correct id to remove notifications of messages that have been read.

File: iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift [hunk 5]
```diff
@@ -58,7 +58,7 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
     /// A reference to the current task that scrolls to a specific message
     private var scrollToMessageTask: Task<Void, Never>?
     /// If true we should scroll to the last item in the collection in layout subviews.
-    private var scrollToLastItemOnLayout: Bool = false
+    private var scrollToFirstUnreadIfNeccessary: Bool = false
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift:61]: Now we attempt to scroll to the first unread message when a conversation is loaded.

File: iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift [hunk 6]
```diff
@@ -97,9 +97,31 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
 
         self.collectionView.expandToSuperviewSize()
 
-        if self.scrollToLastItemOnLayout {
-            self.scrollToLastItemOnLayout = false
-
+        if self.scrollToFirstUnreadIfNeccessary {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift:100]: I just reused this in the layout code.

File: iOS/Flows/Conversation/Cells/MessageCell.swift [hunk 7]
```diff
@@ -148,6 +152,21 @@ class MessageCell: UICollectionViewCell {
...
         self.startConsumptionIfNeeded(for: message)
+        self.updateStatusText(for: message)
+    }
+        
+    private func updateStatusText(for message: Message) {
+        guard message.deliveryStatus == .sent || message.deliveryStatus == .read else { return }
+        
+        self.statusTextTask = Task {
+            
+            self.messageState.statusText = message.context.displayName
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Cells/MessageCell.swift:163]: The status text now displays the context and the time stamp

File: iOS/Flows/People/InvitationLoadingView.swift [hunk 8]
```diff
@@ -13,15 +13,19 @@ import Localization
 class InvitationLoadingView: BaseView {
 
     let blurView = BlurView()
-    let label = ThemeLabel(font: .regular)
+    let label = ThemeLabel(font: .small)
```
Comment (Benjamindodgson) [inline iOS/Flows/People/InvitationLoadingView.swift:16]: Updated layout for loading screen.

File: iOS/UI/Swipeable Input/MessageDeliveryStatusUIViewRepresentable.swift [hunk 9]
```diff
@@ -28,24 +28,23 @@ struct MessageDeliveryStatusUIViewRepresentable: UIViewRepresentable {
 
 class MessageDeliveryStatusUIView: BaseView {
 
-    private let readStatusView = AnimationView(name: "visibility")
-    private let sendStatus = AnimationView(name: "checkmark")
-    private let errorStatusView = AnimationView(name: "alertCircle")
+    private let readStatusView = AnimationView.with(animation: .doubleCheckMark)
```
Comment (Benjamindodgson) [inline iOS/UI/Swipeable Input/MessageDeliveryStatusUIViewRepresentable.swift:31]: using the pre defined enum

File: iOS/UI/Swipeable Input/MessageDeliveryStatusUIViewRepresentable.swift [hunk 10]
```diff
@@ -28,24 +28,23 @@ struct MessageDeliveryStatusUIViewRepresentable: UIViewRepresentable {
...
 
         self.addSubview(self.sendStatus)
         self.sendStatus.currentProgress = 0
         self.sendStatus.setValueProvider(colorProvider, keypath: keypath)
         self.sendStatus.contentMode = .scaleAspectFit
 
         self.addSubview(self.readStatusView)
-        self.readStatusView.currentProgress = 1
-        self.readStatusView.setValueProvider(colorProvider, keypath: keypath)
+        self.readStatusView.currentProgress = 0
```
Comment (Benjamindodgson) [inline iOS/UI/Swipeable Input/MessageDeliveryStatusUIViewRepresentable.swift:47]: The double check mark looks bad but we will just replace it soon.

File: iOS/UI/Swipeable Input/MessageStatusView.swift [hunk 11]
```diff
@@ -33,17 +33,16 @@ private struct MessageDeliveryStatusView: View {
...
-                    .fontType(.small)
-                    .color(.T1)
-
-                Spacer.length(.short)
-            }
+            
+            Text(config.statusText)
+                .fontType(.small)
+                .color(.T1)
+                .animation(.linear(duration: Theme.animationDurationFast), value: config.statusText)
```
Comment (Benjamindodgson) [inline iOS/UI/Swipeable Input/MessageStatusView.swift:43]: This is how you animate stuff in SwiftUI

Comment (Benjamindodgson) [reply iOS/UI/Swipeable Input/MessageStatusView.swift:43]: pretty cool imo

Comment (Benjamindodgson) [reply iOS/UI/Swipeable Input/MessageStatusView.swift:43]: Also responds directly to observable object changing.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**